tests/test_tracing.py

```diff
@@ -184,7 +227,16 @@ class TestTraceCalls:
     def test_simple_call(self, collector):
         with trace_calls(collector, max_typed_dict_size=0):
             simple_add(1, 2)
-        assert collector.traces == [CallTrace(simple_add, {'a': int, 'b': int}, int)]
+        assert collector.traces == [
+            CallTrace(
+                simple_add,
+                {"a": int, "b": int},
+                int,
+                caller=pytestRegex(
+                    f"^{get_cur_filename()}:{get_cur_funcname()}:[0-9]*$"
```

Comment:
this is pretty verbose and repetitive in all these tests. I think we can improve it a lot by defining a wrapper or subclass of `pytestRegex` called like `match_caller` that always adds to the pattern the initial current-filename (which can just be based on `__file__`, you don't need to use frame-walking for this) and the final line number match part, so each test can just provide only the function name, say something like `caller=match_caller("TestTraceCalls.test_simple_call")` (I think I'd also prefer just hardcoding the function name for simplicity rather than using `get_cur_funcname()`)